tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -426,7 +426,9 @@ def process_resource_set(self, resources):
         for resource in resources:
             settings = client.diagnostic_settings.list(resource['id'])
             settings = [s.as_dict() for s in settings.value]
-
+            # put an empty item in when no diag settings, so the absent operator can function
+            if(not settings):
```

Comment:
nit - i'm somewhat surprised this passed flake8 linting, more idiomatic would just be

`if not settings:`

Reply:
fixed